packages/cli/src/services/hooks.service.ts

```diff
@@ -0,0 +1,118 @@
+import { Service } from 'typedi';
+import type { NextFunction, Response } from 'express';
+import type { QueryDeepPartialEntity } from '@n8n/typeorm/query-builder/QueryPartialEntity';
+import type { FindManyOptions, FindOneOptions, FindOptionsWhere } from '@n8n/typeorm';
+
+import { AuthService } from '@/auth/auth.service';
+import type { AuthUser } from '@db/entities/AuthUser';
+import type { User } from '@db/entities/User';
+import { UserRepository } from '@db/repositories/user.repository';
+import { SettingsRepository } from '@db/repositories/settings.repository';
+import { WorkflowRepository } from '@db/repositories/workflow.repository';
+import { CredentialsRepository } from '@db/repositories/credentials.repository';
+import type { WorkflowEntity } from '@db/entities/WorkflowEntity';
+import type { CredentialsEntity } from '@db/entities/CredentialsEntity';
+import type { Settings } from '@db/entities/Settings';
+import { UserService } from '@/services/user.service';
+import type { AuthenticatedRequest } from '@/requests';
+import type { Invitation } from '@/Interfaces';
+
+/**
+ * Exposes functionality to be used by the cloud BE hooks.
+ * DO NOT DELETE or RENAME any of the methods without making sure this is not used in cloud BE hooks.
+ */
+@Service()
+export class HooksService {
+	constructor(
+		private readonly userService: UserService,
+		private readonly authService: AuthService,
+		private readonly userRepository: UserRepository,
+		private readonly settingsRepository: SettingsRepository,
+		private readonly workflowRepository: WorkflowRepository,
+		private readonly credentialsRepository: CredentialsRepository,
+	) {}
+
+	/**
+	 * Invite users to instance during signup
+	 */
+	async inviteUsers(owner: AuthUser, attributes: Invitation[]) {
```

Comment:
@netroy Why did we add `AuthUser` here if the invite users functionality does not use the MFA properties we have in the AuthUser repository?

Reply:
sorry, it was a mistake. fixed it.

Reply:
All good. Thought I was missing something!